### PR TITLE
fix: attach element when used in drag source

### LINKED ISSUE #20426: Support for invisible/detached components for setDragImage
### Describe your motivation

The recently added [Drag Image support](https://github.com/vaadin/flow/issues/6793) is limited to cases, when the component is a visible element in the viewport. The component does not appear during drag, if it's just a detached component:
```java
Div dragBox = new Div("Drag box");
DragSource<Div> dragSource = DragSource.configure(dragBox);
dragSource.setDraggable(true);
dragSource.setDragImage(new Div("Drag Source Image"));
add(dragBox);
```
see [this comment](https://github.com/vaadin/flow/issues/6793#issuecomment-2461532024).

`setDragImage` browser's API states that the element should be visible or added to a DOM, see https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/setDragImage#imgelement.

### Describe the solution you'd like

With adding a component to a DOM and offset it like this (just an example), makes it shown as drag image.
```java
add(dragSourceImage);
dragSourceImage.getStyle().set("position", "absolute");
dragSourceImage.getStyle().set("top", "-100px");
dragSourceImage.getStyle().set("left", "-100px");
```

Vaadin Flow's DnD can integrate some logic that add/hides the component on a page to make the API work.
A bit of hacky solution, There is no other way currently to make it work.

### Describe alternatives you've considered

Document possible workarounds that can be applied in projects, but IMO that's a wrong way.

### Additional contex
Vaadin 24.6 pre-release.

### PR DESCRIPTION
**Pull request by caalador**
Attach elemnt to dom and move it
outside of the viewport to have it
visible as a drag image.
Image can be used without attaching
to dom. Hidden elements are also
not shown so throw exception
when using one as dragImage.

Fixes #20426

### CODE REVIEW

**Review by mshabarov — DISMISSED** [review]
I have no strong opinion should the framework add the component by it's own or delegate this to projects. We have agreed that we just document this. 
However, with this change, the Flow allows to tune how the component is attached.

Thus, I like this change - to have a default and provide an ability to customise it.

File: flow-dnd/src/main/java/com/vaadin/flow/component/dnd/DragSource.java [hunk 1]
```diff
@@ -380,6 +385,20 @@ default void setDragImage(Component dragImage, int offsetX, int offsetY) {
...
     }
 
+    private void appendDragElement(Element dragElement) {
+        if (dragElement.getTag().equals("img")) {
+            getDraggableElement().appendVirtualChild(dragElement);
+        } else {
+            LoggerFactory.getLogger(DragSource.class).info(
+                    "Attaching child to dom in position -100,-100. Consider adding the component manually to not get overlapping components on drag for element.");
+            getDraggableElement().appendChild(dragElement);
+            Style style = dragElement.getStyle();
```
[the remark this answers is not in the document]
Comment (caalador) [reply flow-dnd/src/main/java/com/vaadin/flow/component/dnd/DragSource.java:395]: Done

**Review by tltv — APPROVED** [review]

File: flow-dnd/src/main/java/com/vaadin/flow/component/dnd/DragSource.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mshabarov) [reply flow-dnd/src/main/java/com/vaadin/flow/component/dnd/DragSource.java:395]: This doesn't seem to work. Tested with the following example:
```java
Div dragBox = new Div("Drag box");
DragSource<Div> dragSource = DragSource.create(dragBox);
dragSource.setDraggable(true);
Button button = new Button("Button");
dragSource.setDragImage(button);
add(dragBox);
```

It doesn't show anything.

Though it works once I configure it manually:
```java
Div dragBox = new Div("Drag box");
DragSource<Div> dragSource = DragSource.create(dragBox);
dragSource.setDraggable(true);
Button button = new Button("Button");
add(button);
Element element = button.getElement();
Style style = element.getStyle();
style.set("position", "absolute");
style.set("top", "-100px");
style.set("left", "-100px");
// style.set("display", "none"); //also works even if hidden 
dragSource.setDragImage(button);
add(dragBox);
```

File: flow-dnd/src/main/java/com/vaadin/flow/component/dnd/DragSource.java [hunk 2]
```diff
@@ -380,6 +385,21 @@ default void setDragImage(Component dragImage, int offsetX, int offsetY) {
                 (dragImage == null ? 0 : offsetY), getDraggableElement());
     }
 
+    private void appendDragElement(Element dragElement) {
+        if (dragElement.getTag().equals("img")) {
+            getDraggableElement().appendVirtualChild(dragElement);
+        } else {
+            LoggerFactory.getLogger(DragSource.class).info(
```
Comment (mshabarov) [inline flow-dnd/src/main/java/com/vaadin/flow/component/dnd/DragSource.java:392]:
```suggestion
            LoggerFactory.getLogger(DragSource.class).debug(
```

**Comment by mshabarov** [comment]
Added a workaround to docs, while this is being investigated. https://github.com/vaadin/docs/pull/3936

**Comment by mshabarov** [comment]
I tried doing the same trick using a plain JavaScript and it works as expected for me, the styles are similar and the drag image component is a child of drag source component. I'll go checking what's wrong in Flow because a browser is likely out of the loop here.

setdragimage

**Comment by mshabarov** [comment]
It was my fault, I had 
```css
.v-dragged {
  opacity: 0;
}
```
leftover css in my test project that ofc made things worse.

**Review by mshabarov — APPROVED** [review]

**Pull request merged**